src/collide.cpp

```diff
@@ -1687,7 +1687,7 @@ void Collide::adapt_grid()
 
 void Collide::grow_percell(int n)
 {
-  if (nglocal+n < nglocalmax) return;
+  if (nglocal+n < nglocalmax || !ngroups) return;
   nglocalmax += DELTAGRID;
```

Comment:
Also, I think the 2nd line needs to be:
while (nglocal+n < nglocalmax) nglocalmax += DELTAGRID;